core/src/main/java/org/infinispan/health/ClusterHealth.java

```diff
@@ -0,0 +1,27 @@
+package org.infinispan.health;
+
+/**
+ * Cluster health information.
+ */
+public interface ClusterHealth {
+
+    /**
+     * Returns total cluster health.
+     */
+    HealthStatus getHealthStatus();
+
+    /**
+     * Returns the name of the cluster.
+     */
+    String getClusterName();
+
+    /**
+     * Returns the number of nodes in the cluster.
+     */
+    int getNumberOfNodes();
+
+    /**
+     * Returns node names.
+     */
+    String[] getNodeNames();
```

Comment:
-1 to returning arrays, you can't make them immutable so you always have to copy them.